batch/batch/server/database.py

```diff
@@ -1,6 +1,78 @@
 from ..database import Database, Table, make_where_statement
 
 
+class BatchBuilder:
+    def __init__(self, batch_db, n_jobs, log):
+        self._log = log
+        self._db = batch_db
+        self._conn = None
+        self._batch_id = None
+        self._is_open = True
+        self._jobs = []
+        self._jobs_parents = []
+
+        self._jobs_sql = None
+        self._jobs_parents_sql = None
+
+        self.n_jobs = n_jobs
+
+    def close(self):
+        if self._conn is not None:
+            self._db.pool.release(self._conn)
+            self._conn = None
+        self._is_open = False
+
+    async def create_batch(self, **items):
+        assert self._is_open
+        if self._batch_id is not None:
+            raise ValueError("cannot create batch more than once")
+
+        self._conn = await self._db.pool.acquire()
+        await self._conn.autocommit(False)
+
+        await self._conn.begin()
+
+        sql = self._db.batch.new_record_template(*items)
+        async with self._conn.cursor() as cursor:
+            await cursor.execute(sql, dict(items))
+            self._batch_id = cursor.lastrowid
+        return self._batch_id
+
+    def create_job(self, **items):
+        assert self._is_open
+        if len(self._jobs) == 0:
+            self._jobs_sql = self._db.jobs.new_record_template(*items)
```

Comment:
It doesn't seem right that `create_job` will break if it is called twice with different sets of kwargs. Are all the arguments required? Should we make that explicit?
```
_job_params = set(['batch_id', ...])
def create_job(self, **items):
    assert set(items) == BatchBuilder._job_params
    if len(self._jobs) == 0:
        self._jobs_sql = self._db.jobs.new_record_template(*BatchBuild._job_params)
    ....
```